providers/linux/util.go

```diff
@@ -26,20 +26,40 @@ import (
 	"strconv"
 )
 
-func parseKeyValue(content []byte, separator string, callback func(key, value []byte) error) error {
-	sc := bufio.NewScanner(bytes.NewReader(content))
-	for sc.Scan() {
-		parts := bytes.SplitN(sc.Bytes(), []byte(separator), 2)
-		if len(parts) != 2 {
+func parseKeyValue(content []byte, separator byte, callback func(key, value []byte) error) error {
+	for len(content) > 0 {
+		sepIdx := bytes.IndexByte(content, separator)
```

Comment:
Should we tolerate empty lines? The previous code did.

Reply:
added a test specifically for empty lines